sailthru_content/services/sailthru_translation_service.py

```diff
@@ -0,0 +1,275 @@
+            sailthru_content_vars['programs'] = []
+            for program_link in course_programs:
+                program = program_dictionary.get(program_link.get('uuid'))
+                if program:
+                    sailthru_content_vars['programs'].append(self._translate_program(program))
+
+        return sailthru_content_vars
+
+    def _get_tags_from_property(self, source_array, key, tag_prefix):
+        tags = []
+        if source_array:
+            for tag_item in source_array:
+                tags.append(self._convert_tag(tag_prefix, tag_item.get(key)))
+
+        return tags
+
+    def translate_course_run(self, course_run, course, program_dictionary=None):
+        # get marketing url
+        url = course_run['marketing_url']
+        if not url:
+            url = course['marketing_url']
+
+        # create parameters for call to Sailthru
+        sailthru_content = {}
+        sailthru_content['url'] = '{}/courses/{}/info'.format(self.lms_root, course_run['key'])
+        sailthru_content['title'] = course_run['title']
+        if course_run['short_description']:
+            sailthru_content['description'] = course_run['short_description']
+
+        # get first owner
+        if course.get('owners') and len(course.get('owners')) > 0:
+            sailthru_content['site_name'] = course['owners'][0]['key'].replace('_', ' ')
+
+        # use last modified date for sailthru 'date'
+        sailthru_content['date'] = self._convert_date(course_run.get('modified'))
+
+        # use enrollment_end for sailthru expire_date
+        if course_run.get('enrollment_end'):
+            sailthru_content['expire_date'] = self._convert_date(course_run['enrollment_end'])
+        elif course_run.get('end'):
+            sailthru_content['expire_date'] = self._convert_date(course_run['end'])
+
+        # get the image, if any
+        if course_run.get('image') and course_run['image']['src']:
+            sailthru_content['images'] = {'thumb': {'url': course_run['image']['src']}}
+
+        # create the interest tags
+        tags = self._get_tags_from_property(course.get('subjects'), 'name', 'subject')
+        tags.extend(self._get_tags_from_property(course.get('owners'), 'key', 'school'))
+        tags.extend(self._get_tags_from_property(course.get('sponsors'), 'key', 'school'))
+
+        if len(tags) > 0:
+            sailthru_content['tags'] = ", ".join(tags)
+
+        sailthru_content['vars'] = self._create_course_vars(
+            course,
+            course_run,
+            url,
+            sailthru_content.get('site_name'),
+            program_dictionary,
+        )
+        sailthru_content['spider'] = 0
+
+        # perform any fixups
+        if self.fixups:
+            for row in self.fixups:
+                if row[0] == course_run['key']:
+                    if row[1] == 'var':
+                        logging.info('Changing var.%s to %s for %s', row[2], row[3], row[0])
```

Comment:
Agree. However, the fixups are a temporary measure that I hope to remove in the near future. I don't think it would worth the effort to make this prettier now